api/routes/settings.ts

```diff
@@ -0,0 +1,70 @@
+import { FastifyPluginCallback } from 'fastify';
+import { Type } from '@sinclair/typebox';
+
+export const settingRoutes: FastifyPluginCallback = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.put<{
+    Body: {
+      profileUI: {
+        isLocked: boolean;
+        showAbout: boolean;
+        showCerts: boolean;
+        showDonation: boolean;
+        showHeatMap: boolean;
+        showLocation: boolean;
+        showName: boolean;
+        showPoints: boolean;
+        showPortfolio: boolean;
+        showTimeLine: boolean;
+      };
+    };
+  }>(
+    '/update-my-profileui',
+    {
+      schema: {
+        body: Type.Object({
+          profileUI: Type.Object({
+            isLocked: Type.Boolean(),
+            showAbout: Type.Boolean(),
+            showCerts: Type.Boolean(),
+            showDonation: Type.Boolean(),
+            showHeatMap: Type.Boolean(),
+            showLocation: Type.Boolean(),
+            showName: Type.Boolean(),
+            showPoints: Type.Boolean(),
+            showPortfolio: Type.Boolean(),
+            showTimeLine: Type.Boolean()
+          })
```

Comment:
Could you add a response schema here? It should include 200 and 500 values. Ideally, since they'll be common to a lot of responses, could you put them in the types dir?

The schemas in question are

```ts
          200: Type.Object({
            message: Type.String(),
            type: Type.Literal('success')
          }),
          500: Type.Object({
            message: Type.String(),
            type: Type.Literal('danger')
          })
```